src/Symfony/Component/Translation/Catalogue/AbstractOperation.php

```diff
@@ -83,7 +83,19 @@ public function __construct(MessageCatalogueInterface $source, MessageCatalogueI
     public function getDomains()
     {
         if (null === $this->domains) {
-            $this->domains = array_values(array_unique(array_merge($this->source->getDomains(), $this->target->getDomains())));
+            $getAllDomains = static function (MessageCatalogueInterface $catalogue) {
+                $domains = $catalogue->getDomains();
+
+                foreach ($domains as $domain) {
+                    if ([] !== $catalogue->all($domainIcu = $domain.MessageCatalogueInterface::INTL_DOMAIN_SUFFIX)) {
+                        $domains[] = $domainIcu;
```

Comment:
Appending values to an array we are iterating on may be confusing for future contributions here. Shouldn't we use 2 different arrays `$domains` (for foreach) and `$allDomains` (for collection)?

Reply:
Why not, if it can improve the readability I'm in favour of that!